src/opm/input/eclipse/Schedule/Well/WellInjectionProperties.cpp

```diff
@@ -130,6 +132,12 @@ namespace Opm {
                 throw std::invalid_argument("Tried to set invalid control: " + cmodeString + " for well: " + well_name);
             }
         }
+        this->rsRvInj = record.getItem("VAPOIL_C").getSIDouble(0);
+        if (this->injectorType == InjectorType::OIL && this->rsRvInj > 0) {
+            double factor = record.getItem("VAPOIL_C").getSIDouble(0) / record.getItem("VAPOIL_C").get<double>(0);
+            // for oil injectors the rsRvInj is Rs and the inverse of the si-conversion factor should be used
+            this->rsRvInj =  record.getItem("VAPOIL_C").get<double>(0) / factor;
```

Comment:
This is one of this very nice keywords items where the unit depends on another item. If the injector type is oil the item gives the amount of gas (Rs) in the injected oil if the injector type is gas the item gives the amount of oil (Rv) in the injected gas. A very simple way to handle the difference in units is to use the fact that the unit of Rs and Rv are each others inverse. In the keyword description the unit is set to "OilDissolutionFactor" so that will give the correct conversion for Gas injector but for Oil injectors we need to use the 1/factor to get the correct unit conversion.

Reply:
@blattms Are you happy with my answer? This is just a simple extension of already existing functionality. I have tested it and it works. If no more comments, I will merge this later today.